lib/ecto/exceptions.ex

```diff
@@ -117,3 +117,18 @@ end
 defmodule Ecto.MigrationError do
   defexception [:message]
 end
+
+defmodule Ecto.StaleModelError do
+  defexception [:message]
+
+  def exception(opts) do
+    action = Keyword.fetch!(opts, :action)
+    model = Keyword.fetch!(opts, :model)
+
+    msg = """
+    attempted to #{action} a stale #{model}
```

Comment:
Oh, sorry. Definitely pass the whole model (the struct) and print it all. You usually want all info (like lock version) when those things happen.